packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -1182,7 +1182,12 @@ export const mixinSummaryHandler = (
 
 		async summarize(...args: any[]) {
 			const summary = await super.summarize(...args);
-			const content = await handler(this);
+
+			// Any error coming from app-provided handler should be marked as DataProcessingError
+			const content = await handler(this).catch((e: unknown) => {
+				throw DataProcessingError.wrapIfUnrecognized(e, "mixinSummaryHandler");
+			});
```

Comment:
Nit: Would it be better to wrap this in a try catch statement? Doesn't really matter to me. Feel free to ignore

Reply:
You know what?  It _is_ better to wrap in a try-catch.  As it is, if the `handler` passed in is a non-`async` function and it throws, the error will escape this wrapping.  This tripped me up when testing, because that's exactly what I was passing.  But in practice, that would be a very strange to write that callback. And would likely trigger the `promise-function-async` lint rule.  However, as my test proved, that rule has a gap (I didn't get the lint error).

I personally like the `.catch` style better here, but since you mentioned it, and there is a theoretical gap, I will switch it.

Reply:
It's also good because now the call to `addBlob` is covered too, which includes some input validation (the output of `handler`)